app/controllers/steps_controller.rb

```diff
@@ -0,0 +1,71 @@
+class StepsController < ApplicationController
+  helper FormElementsHelper
+
+  def index
+    @prisoner_step = PrisonerStep.new
+    render :prisoner_step
+  end
+
+  def create
+    render select_step
+  end
+
+private
+
+  STEPS = %i[
+    prisoner_step
+    visitors_step
+    slots_step
+    confirmation_step
+  ]
+
+  def select_step
+    STEPS.find { |s| send(s) } || complete
+  end
```

Comment:
I'm not really happy with the way this mixes up the functional and the side-effectful, but all my attempted refactors have been even worse.